upf/expression.py

```diff
@@ -171,18 +188,61 @@ def Bool(self, value: bool) -> FNode:
         else:
             return self.false_expression
 
-    def FluentExp(self, fluent: 'upf.Fluent', params: Tuple[Expression, ...] = tuple()) -> FNode:
-        """ Creates an expression for the given fluent and parameters.
-        Restriction: parameters type must be compatible with the fluent signature
+    def Int(self, value: int) -> FNode:
+        """Return an int constant."""
+        if type(value) != int:
+            raise UPFTypeError("Expecting int, got %s" % type(value))
+        return self.create_node(node_type=op.INT_CONSTANT, args=tuple(), payload=value)
+
+    def Real(self, value: Fraction) -> FNode:
+        """Return a real constant."""
+        if type(value) != Fraction:
+            raise UPFTypeError("Expecting Fraction, got %s" % type(value))
+        return self.create_node(node_type=op.REAL_CONSTANT, args=tuple(), payload=value)
+
+    def Plus(self, left: Expression, right: Expression) -> FNode:
+        """ Creates an expression of the form:
+            left + right
         """
-        assert fluent.arity() == len(params)
-        params_exp = self.auto_promote(*params)
-        return self.create_node(node_type=op.FLUENT_EXP, args=tuple(params_exp), payload=fluent)
+        [left, right] = self.auto_promote(left, right)
+        return self.create_node(node_type=op.PLUS, args=(left, right))
 
-    def ParameterExp(self, param: 'upf.ActionParameter') -> FNode:
-        """Returns an expression for the given action parameter."""
-        return self.create_node(node_type=op.PARAM_EXP, args=tuple(), payload=param)
+    def Minus(self, left: Expression, right: Expression) -> FNode:
+        """ Creates an expression of the form: left - right."""
+        [left, right] = self.auto_promote(left, right)
+        return self.create_node(node_type=op.MINUS, args=(left, right))
 
-    def ObjectExp(self, obj: 'upf.Object') -> FNode:
-        """Returns an expression for the given object."""
-        return self.create_node(node_type=op.OBJECT_EXP, args=tuple(), payload=obj)
+    def Times(self, left: Expression, right: Expression) -> FNode:
+        """ Creates an expression of the form: left * right."""
+        [left, right] = self.auto_promote(left, right)
```

Comment:
No need to create a list:
```suggestion
        left, right = self.auto_promote(left, right)
```